Source/DafnyCore/Dafny.atg

```diff
@@ -3202,6 +3203,52 @@ Forall = "forall".
 Exists = "exists".
 QSep = "::".
 
+Expression<out Expression e, bool allowLemma, bool allowLambda, bool allowBitwiseOps = true>
+=
+  NotDecreasesToExpression<out e, allowLemma, allowLambda, allowBitwiseOps>
+  .
+
+MaybeDecreasesToExpression<out Expression e, IToken lp>
+= (. List<ActualBinding> args = new();
+     List<Expression> newExprs = new();
+     Expression tmp = null;
+     IToken x = null;
+     bool allowNoChange = false;
+  .)
+  TupleArgs<args>
+  [ ( "decreases"                                                               (. allowNoChange = false; .)
+    | "nonincreases"                                                            (. allowNoChange = true; .)
```

Comment:
Have you considered `not` instead of `non` ? Does `non` occur anywhere else? `Not` is what I first think of for negation.

Reply:
"nondecreasing" is a word, but "notdecreasing" isn't